### PR TITLE
Create GoogleMediaImporter

### PR DESCRIPTION
**Pull request by seehamrun**
This is based off of PR #1006 which pulled out some of the reusable functionality from the GooglePhotosImporter. The GoogleMediaImporter is a copy of that file, but with 2 additional functions: `importVidoes` and `importVideoBatch`

Once PR #1006 is submitted, the diff should be cleaner

### CODE REVIEW

**Comment by seehamrun** [comment]
Ok i have rebased this branch off of PR #1004 so the history got removed, but it is still a copy of the GooglePhotosImporter w/  2 functions added `importVidoes`, `importVideoBatch`

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/MediaResult.java [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 import java.io.Serializable;
 
-class PhotoResult implements Serializable {
+public class MediaResult implements Serializable {
```
Comment (jzacsh) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/MediaResult.java:5]: nit: should this file land in the google/media/ folder instead?

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/media/GoogleMediaImporter.java [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+    // Uploads album metadata
+    if (data.getAlbums() != null && data.getAlbums().size() > 0) {
+      for (MediaAlbum album : data.getAlbums()) {
+        idempotentImportExecutor.executeAndSwallowIOExceptions(
+            album.getId(), album.getName(), () -> importSingleAlbum(jobId, authData, album));
+      }
+    }
+
+    long bytes = importPhotos(data.getPhotos(), idempotentImportExecutor, jobId, authData);
+    bytes += importVideos(data.getVideos(), idempotentImportExecutor, jobId, authData);
```
Comment (alexeyqu-fb) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/media/GoogleMediaImporter.java:134]: We assume that `MediaContainerResource` has either photos or videos and process them at the same time. 
Is it possible that a single `MediaContainerResource` has both?

**Review by alexeyqu-fb — COMMENTED** [review]
The proposed `GoogleMediaImporter` has pairs of identical methods like `importPhotoBatch`, `importVideoBatch` etc. 
I am not sure I get why do we want to merge it into a common class -- is there a future where photo and video transfers would be merged?

I would rather steer towards having generic `PhotoImporter` and `VideoImporter` that define the API for every other adapter to inherit from, define the common stuff there (counts/bytes reporting, common ways to report errors etc), and then overriding the exact destination-specific import instructions (which API to call, batching, etc) on the `DestinationPhotoImporter` level.

**Pull request closed without merging**